iam-login-service/src/main/java/it/infn/mw/iam/core/oauth/IamOAuthConfirmationController.java

```diff
@@ -195,7 +207,7 @@ public String confimAccess(Map<String, Object> model,
     }
 
     // add in any scopes that aren't system scopes to the end of the list
-    sortedScopes.addAll(Sets.difference(scopes, systemScopes));
+    sortedScopes.addAll(Sets.difference(scopeService.fromStrings(filteredScopes), systemScopes));
```

Comment:
Because `filteredScopes` is a `Set<String>` and we need to convert it to a `Set<SystemScope>`.
Also, I think it was wrong before, because here we should consider the difference between the filtered scopes (not `scopes`, which are the scopes in the authorization request) and the system scopes to sort them and show them in the content page (put the system scopes first and then the others).

Reply:
Ok, this fix doesn't affect system scope logic (or filter) but eventually avoids policy on custom scopes (can these exist?) to be ignored or bypassed